app/browser/reducers/ledgerReducer.js

```diff
@@ -335,6 +323,85 @@ const ledgerReducer = (state, action, immutableAction) => {
         state = ledgerState.saveQRCode(state, action.get('currency'), action.get('image'))
         break
       }
+    case appConstants.APP_IDLE_STATE_CHANGED:
+      {
+        if (!getSetting(settings.PAYMENTS_ENABLED)) {
+          break
+        }
+
+        if (action.has('idleState') && action.get('idleState') !== 'active') {
+          state = ledgerApi.pageDataChanged(state)
+        }
+        break
+      }
+    case windowConstants.WINDOW_SET_FOCUSED_FRAME:
+      {
+        if (!getSetting(settings.PAYMENTS_ENABLED)) {
+          break
+        }
+
+        if (action.get('location')) {
+          state = ledgerApi.pageDataChanged(state, {
+            location: action.get('location'),
+            tabId: action.get('tabId')
+          })
+        }
+        break
+      }
+    case appConstants.APP_WINDOW_BLURRED:
+      {
+        if (!getSetting(settings.PAYMENTS_ENABLED)) {
+          break
+        }
+
+        let windowCount = BrowserWindow.getAllWindows().filter((win) => win.isFocused()).length
+        if (windowCount === 0) {
+          state = ledgerApi.pageDataChanged(state, {}, true)
+        }
+        break
+      }
+    case 'event-set-page-info':
+      {
+        if (!getSetting(settings.PAYMENTS_ENABLED)) {
+          break
+        }
+
+        state = ledgerApi.pageDataChanged(state, {
+          location: action.getIn(['pageInfo', 'url'])
+        })
+        break
+      }
+    case appConstants.APP_CLOSE_WINDOW:
+      {
+        if (!getSetting(settings.PAYMENTS_ENABLED)) {
+          break
+        }
+
+        state = ledgerApi.pageDataChanged(state)
+        break
+      }
+    case windowConstants.WINDOW_GOT_RESPONSE_DETAILS:
+      {
+        if (!getSetting(settings.PAYMENTS_ENABLED)) {
+          break
+        }
+
+        // Only capture response for the page (not sub resources, like images, JavaScript, etc)
+        if (action.getIn(['details', 'resourceType']) === 'mainFrame') {
+          const pageUrl = action.getIn(['details', 'newURL'])
+
+          // create a page view event if this is a page load on the active tabId
+          const lastActiveTabId = pageDataState.getLastActiveTabId(state)
+          const tabId = action.get('tabId')
+          if (!lastActiveTabId || tabId === lastActiveTabId) {
+            state = ledgerApi.pageDataChanged(state, {
+              location: pageUrl,
+              tabId: tabId
```

Comment:
nit on style:
just `tabId` instead of `tabId: tabId`

Reply:
done